component/org.wso2.carbon.identity.casque.authenticator/src/main/java/org/wso2/carbon/identity/casque/authenticator/authenticator/CasqueAuthenticator.java

```diff
@@ -0,0 +1,253 @@
+import org.wso2.carbon.identity.application.authentication.framework.exception.LogoutFailedException;
+import org.wso2.carbon.identity.application.authentication.framework.model.AuthenticatedUser;
+import org.wso2.carbon.identity.application.authentication.framework.util.FrameworkConstants;
+import org.wso2.carbon.user.api.UserStoreManager;
+import org.wso2.carbon.identity.core.util.IdentityTenantUtil;
+import org.wso2.carbon.identity.application.common.model.User;
+import org.wso2.carbon.identity.casque.authenticator.constants.CasqueAuthenticatorConstants;
+import org.wso2.carbon.identity.casque.authenticator.authenticator.internal.CasqueAuthenticatorServiceDataHolder;
+import org.wso2.carbon.identity.casque.authenticator.exception.CasqueException;
+import org.wso2.carbon.identity.casque.authenticator.authenticator.radius.Radius;
+import org.wso2.carbon.identity.casque.authenticator.authenticator.radius.RadiusResponse;
+
+import java.util.HashMap;
+import java.util.Map;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class CasqueAuthenticator extends AbstractApplicationAuthenticator implements LocalApplicationAuthenticator {
+
+    private static final long serialVersionUID = 4341535155455223654L;
+    private static final Log log = LogFactory.getLog(CasqueAuthenticator.class);
+    private static final String CASQUE_SNR_CLAIM = "http://wso2.org/claims/identity/casqueSnrToken";
+    private static final String TOKEN_ID_FORMAT = "^[a-fA-F0-9]{3} [0-9]{6}$";  // e.g. "FFF 000001"
+    private static final AuthPages authPages = new AuthPages();
+
+    /**
+     * Getting tokenId from the UserStoreManager.
+     *
+     * @param userName Username of the user
+     * @return true if token is valid otherwise false
+     * @throws CasqueException UserRealm for user or tenant domain is null
+     */
+    private String getCasqueTokenId(String userName) throws CasqueException {
+
+        try {
+            UserStoreManager userStoreManager = CasqueAuthenticatorServiceDataHolder.getInstance()
+                    .getRealmService().getTenantUserRealm(IdentityTenantUtil.getTenantIdOfUser(userName))
+                    .getUserStoreManager();
+
+            // Getting the Token ID assigned to userName
+            Map<String, String> tokenIdMap = userStoreManager.getUserClaimValues(userName,
+                    new String[]{CASQUE_SNR_CLAIM}, null);
+
+            if (tokenIdMap == null || tokenIdMap.get(CASQUE_SNR_CLAIM) == null) {
+                throw new CasqueException("Token ID is null for user: " + userName);
+            }
+            String tokenId = tokenIdMap.get(CASQUE_SNR_CLAIM);
+            if (tokenId.matches(TOKEN_ID_FORMAT)) {
+                return tokenId;
+            }
+            throw new CasqueException(tokenId + "is a bad formatted Token ID for user :" + userName);
```

Comment:
Please put space  as in bellow.
throw new CasqueException(tokenId + " is a bad formatted Token ID for user :" + userName);

Reply:
fixed